sharding/indexHashedNodesCoordinator.go

```diff
@@ -17,48 +17,56 @@ type indexHashedNodesCoordinator struct {
 	nodesMap                map[uint32][]Validator
 	shardConsensusGroupSize int
 	metaConsensusGroupSize  int
+	selfPubKey              []byte
 }
 
 // NewIndexHashedNodesCoordinator creates a new index hashed group selector
-func NewIndexHashedNodesCoordinator(
-	shardConsensusGroupSize int,
-	metaConsensusGroupSize int,
-	hasher hashing.Hasher,
-	shardId uint32,
-	nbShards uint32,
-	nodes map[uint32][]Validator,
-) (*indexHashedNodesCoordinator, error) {
-	if shardConsensusGroupSize < 1 || metaConsensusGroupSize < 1 {
-		return nil, ErrInvalidConsensusGroupSize
+func NewIndexHashedNodesCoordinator(arguments ArgNodesCoordinator) (*indexHashedNodesCoordinator, error) {
+	err := checkArguments(arguments)
+	if err != nil {
+		return nil, err
 	}
 
-	if nbShards < 1 {
-		return nil, ErrInvalidNumberOfShards
+	ihgs := &indexHashedNodesCoordinator{
+		nbShards:                arguments.NbShards,
+		shardId:                 arguments.ShardId,
+		hasher:                  arguments.Hasher,
+		nodesMap:                make(map[uint32][]Validator),
+		shardConsensusGroupSize: arguments.ShardConsensusGroupSize,
+		metaConsensusGroupSize:  arguments.MetaConsensusGroupSize,
+		selfPubKey:              arguments.SelfPublicKey,
 	}
 
-	if shardId >= nbShards && shardId != MetachainShardId {
-		return nil, ErrInvalidShardId
+	err = ihgs.SetNodesPerShards(arguments.Nodes)
+	if err != nil {
+		return nil, err
 	}
 
-	if hasher == nil {
-		return nil, ErrNilHasher
+	return ihgs, nil
+}
+
+func checkArguments(arguments ArgNodesCoordinator) error {
+	if arguments.ShardConsensusGroupSize < 1 || arguments.MetaConsensusGroupSize < 1 {
+		return ErrInvalidConsensusGroupSize
 	}
 
-	ihgs := &indexHashedNodesCoordinator{
-		nbShards:                nbShards,
-		shardId:                 shardId,
-		hasher:                  hasher,
-		nodesMap:                make(map[uint32][]Validator),
-		shardConsensusGroupSize: shardConsensusGroupSize,
-		metaConsensusGroupSize:  metaConsensusGroupSize,
+	if arguments.NbShards < 1 {
+		return ErrInvalidNumberOfShards
```

Comment:
Please add some unit tests to cover all errors in `checkArguments`

Reply:
Done. New unit tests was added.